File: simple-java/src/main/java/com/jy/simple2/AppDatabase.java
package com.jy.simple2;

import com.jy.litedb.annotation.Database;
import com.jy.litedb.api.DBConfig;
import com.jy.litedb.api.LiteDatabase;
import com.jy.litedb.api.YLite;

@Database(entities = {TestJava.class}, name = "app.db", version = 1)
public abstract class AppDatabase extends LiteDatabase {

    abstract public TestJavaDao getTestJavaDao();

    public static synchronized AppDatabase getInstance() {
        return AcPermissionHolder.instance;
    }

    private static class AcPermissionHolder {
        private static final AppDatabase instance = YLite.databaseBuilder(
                MyApplication.getInstance().getApplicationContext(), AppDatabase.class)
                .setOpenDexFileLoader(true)
                .setDBConfig(DBConfig.Companion.beginBuilder().setOpenCache(true).build())
                .build();
    }
}

File: app/src/main/java/com/jy/simple/db/AppDatabase.kt
package com.jy.simple.db

import com.jy.litedb.annotation.Database
import com.jy.litedb.api.DBConfig.Companion.beginBuilder
import com.jy.litedb.api.LiteDatabase
import com.jy.litedb.api.YLite
import com.jy.simple.MyApplication
import com.jy.simple.bean.SimpleInfo
import com.jy.simple.bean.TestInfo

@Database(entities = [TestInfo::class, SimpleInfo::class], name = "app.db", version = 3)
abstract class AppDatabase : LiteDatabase() {
    abstract fun getTestJavaDao(): TestInfoDao

    private object AcPermissionHolder {
        val instance = YLite.databaseBuilder(
            MyApplication.getInstance().applicationContext,
            AppDatabase::class.java
        )
            .setOpenDexFileLoader(true)
            .setDBConfig(beginBuilder().setOpenCache(true).build())
            .build()
    }

    companion object {
        @get:Synchronized
        val instance: AppDatabase
            get() = AcPermissionHolder.instance
    }
}